lib/IRGen/GenMeta.cpp

```diff
@@ -1361,7 +1366,50 @@ namespace {
       }
     }
 
-    void addMethodOverride(SILDeclRef baseRef, SILDeclRef declRef) {}
+    void addOverrideTable() {
+      if (OverrideTableEntries.empty())
+        return;
+
+      B.addInt32(OverrideTableEntries.size());
+
+      for (auto pair : OverrideTableEntries)
+        emitMethodOverrideDescriptor(pair.first, pair.second);
+    }
+
+    void emitMethodOverrideDescriptor(SILDeclRef baseRef, SILDeclRef declRef) {
+      auto descriptor = B.beginStruct(IGM.MethodOverrideDescriptorStructTy);
+
+      // The class containing the base method.
+      auto *baseClass = cast<ClassDecl>(baseRef.getDecl()->getDeclContext());
+      auto baseClassEntity = LinkEntity::forNominalTypeDescriptor(baseClass);
+      auto baseClassDescriptor =
+        IGM.getAddrOfLLVMVariableOrGOTEquivalent(
+          baseClassEntity, IGM.getPointerAlignment(),
+          IGM.Int8Ty);
+      descriptor.addRelativeAddress(baseClassDescriptor);
+
+      // The base method.
+      auto baseMethodEntity = LinkEntity::forMethodDescriptor(baseRef);
+      auto baseMethodDescriptor =
+        IGM.getAddrOfLLVMVariableOrGOTEquivalent(
+          baseMethodEntity, IGM.getPointerAlignment(),
```

Comment:
Method descriptors are definitely not pointer-aligned.  I don't think the alignment on the LLVM global will actually change anything here, but still, we shouldn't make promises we can't keep.

Reply:
Good point.